priam/src/main/java/com/netflix/priam/backupv2/IMetaProxy.java

```diff
@@ -78,7 +78,7 @@ public interface IMetaProxy {
      * @param dateRange the time period to scan in the remote file system for incremental files.
      * @return iterator containing the list of path on the remote file system satisfying criteria.
      */
-    Iterator<AbstractBackupPath> getIncrementals(DateUtil.DateRange dateRange);
+    ImmutableList<AbstractBackupPath> getIncrementals(DateUtil.DateRange dateRange);
```

Comment:
mostly curious: why this change? the iterator gives us flexibility that the data could be "streamed" where as this locks us in to determining all the incrementals up front. Likely not an issue but something I noticed.

Reply:
Looking at the implementation we have converted from something that would get called on each iteration to something that gets called all during the execution of this method. Not sure thats a problem or not but wanted to call it out.